File: Android/app/src/main/java/com/agora/entfulldemo/welcome/UserAgreementDialog.java
package com.agora.entfulldemo.welcome;

import android.content.Context;
import android.os.Build;
import android.text.Spannable;
import android.text.SpannableStringBuilder;
import android.text.method.LinkMovementMethod;
import android.text.style.ClickableSpan;
import android.text.style.ForegroundColorSpan;
import android.view.Gravity;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.Window;
import android.view.WindowManager;
import androidx.annotation.NonNull;
import androidx.core.content.ContextCompat;
import com.agora.entfulldemo.databinding.AppDialogUserAgreementBinding;
import io.agora.scene.base.component.BaseDialog;
import io.agora.scene.base.manager.PagePilotManager;
import io.agora.scene.widget.R;

public class UserAgreementDialog extends BaseDialog<AppDialogUserAgreementBinding> {
    public UserAgreementDialog(@NonNull Context context) {
        super(context);
    }

    @NonNull
    @Override
    protected AppDialogUserAgreementBinding getViewBinding(@NonNull LayoutInflater inflater) {
        return AppDialogUserAgreementBinding.inflate(inflater);
    }

    @Override
    protected void initView() {
        setCancelable(false);
        getBinding().btnDisagree.setOnClickListener(view -> {
            getOnButtonClickListener().onLeftButtonClick();
        });

        ClickableSpan protocolClickableSpan1 = new ClickableSpan() {
            @Override
            public void onClick(@NonNull View widget) {
                PagePilotManager.pageWebView("https://accktvpic.oss-cn-beijing.aliyuncs.com/pic/meta/demo/fulldemoStatic/privacy/service.html");
            }
        };
        ClickableSpan protocolClickableSpan2 = new ClickableSpan() {
            @Override
            public void onClick(@NonNull View widget) {
                PagePilotManager.pageWebView("https://accktvpic.oss-cn-beijing.aliyuncs.com/pic/meta/demo/fulldemoStatic/privacy/privacy.html");
            }
        };
        ClickableSpan protocolClickableSpan3 = new ClickableSpan() {
            @Override
            public void onClick(@NonNull View widget) {
                PagePilotManager.pageWebView("https://accktvpic.oss-cn-beijing.aliyuncs.com/pic/meta/demo/fulldemoStatic/privacy/service.html");
            }
        };
        ClickableSpan protocolClickableSpan4 = new ClickableSpan() {
            @Override
            public void onClick(@NonNull View widget) {
                PagePilotManager.pageWebView("https://accktvpic.oss-cn-beijing.aliyuncs.com/pic/meta/demo/fulldemoStatic/privacy/privacy.html");
            }
        };

        ForegroundColorSpan spanColor = new ForegroundColorSpan(ContextCompat.getColor(getContext(), R.color.blue_9F));
        SpannableStringBuilder spannable = new SpannableStringBuilder(getContext().getString(R.string.privacy_protection_tip1));
        spannable.setSpan();

        spannable.setSpan(protocolClickableSpan1, 72, 76, Spannable.SPAN_EXCLUSIVE_EXCLUSIVE);
        spannable.setSpan(spanColor, 72, 76, Spannable.SPAN_EXCLUSIVE_EXCLUSIVE);

        spannable.setSpan(protocolClickableSpan2, 79, 83, Spannable.SPAN_EXCLUSIVE_EXCLUSIVE);
        spannable.setSpan(spanColor, 79, 83, Spannable.SPAN_EXCLUSIVE_EXCLUSIVE);

        spannable.setSpan(protocolClickableSpan3, 202, 207, Spannable.SPAN_EXCLUSIVE_EXCLUSIVE);
        spannable.setSpan(spanColor, 202, 207, Spannable.SPAN_EXCLUSIVE_EXCLUSIVE);

        spannable.setSpan(protocolClickableSpan4, 210, 214, Spannable.SPAN_EXCLUSIVE_INCLUSIVE);
        spannable.setSpan(spanColor, 210, 214, Spannable.SPAN_EXCLUSIVE_INCLUSIVE);

        getBinding().tvProtection.setText(spannable);
        getBinding().btnAgree.setOnClickListener(view -> {
            getOnButtonClickListener().onRightButtonClick();
        });
        getBinding().tvProtection.setMovementMethod(LinkMovementMethod.getInstance());
    }

    @Override
    protected void setGravity() {
        Window window = getWindow();
        if (window != null) {
            window.setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN, WindowManager.LayoutParams.FLAG_FULLSCREEN);
            window.setLayout(
                    ViewGroup.LayoutParams.MATCH_PARENT,
                    ViewGroup.LayoutParams.MATCH_PARENT
            );
        }
        getWindow().getAttributes().gravity = Gravity.CENTER;
    }
}

File: Android/app/src/main/java/com/agora/entfulldemo/welcome/UserAgreementDialog.kt
package com.agora.entfulldemo.welcome

import android.content.Context
import android.graphics.Typeface
import android.text.Spannable
import android.text.SpannableStringBuilder
import android.text.TextUtils
import android.text.method.LinkMovementMethod
import android.text.style.ClickableSpan
import android.text.style.ForegroundColorSpan
import android.text.style.LeadingMarginSpan
import android.text.style.StyleSpan
import android.view.*
import androidx.core.content.ContextCompat
import com.agora.entfulldemo.R
import com.agora.entfulldemo.databinding.AppDialogUserAgreementBinding
import com.agora.entfulldemo.home.constructor.URLStatics
import io.agora.scene.base.component.BaseDialog
import io.agora.scene.base.manager.PagePilotManager

class UserAgreementDialog(context: Context) : BaseDialog<AppDialogUserAgreementBinding>(context) {
    override fun getViewBinding(inflater: LayoutInflater): AppDialogUserAgreementBinding {
        return AppDialogUserAgreementBinding.inflate(inflater)
    }

    override fun initView() {
        setCancelable(false)
        binding.btnDisagree.setOnClickListener { view -> getOnButtonClickListener().onLeftButtonClick() }
        val spanColor = ForegroundColorSpan(ContextCompat.getColor(context, R.color.blue_9F))
        // 1
        val spannable1 =
            SpannableStringBuilder(context.getString(R.string.app_agreement_protection_tip1))
        // 2
        val spannable2 =
            SpannableStringBuilder(context.getString(R.string.app_agreement_protection_tip2))
        val protocolClickableSpan1: ClickableSpan = object : ClickableSpan() {
            override fun onClick(widget: View) {
                PagePilotManager.pageWebView(URLStatics.userAgreementURL)
            }
        }
        val protocolClickableSpan2: ClickableSpan = object : ClickableSpan() {
            override fun onClick(widget: View) {
                PagePilotManager.pageWebView(URLStatics.privacyAgreementURL)
            }
        }
        spannable2.setSpan(protocolClickableSpan1, 7, 13, Spannable.SPAN_EXCLUSIVE_EXCLUSIVE)
        spannable2.setSpan(spanColor, 7, 13, Spannable.SPAN_EXCLUSIVE_EXCLUSIVE)
        spannable2.setSpan(protocolClickableSpan2, 14, 20, Spannable.SPAN_EXCLUSIVE_EXCLUSIVE)
        spannable2.setSpan(spanColor, 14, 20, Spannable.SPAN_EXCLUSIVE_EXCLUSIVE)
        spannable2.setSpan(
            LeadingMarginSpan.Standard(0, 32),
            0,
            spannable2.length,
            Spannable.SPAN_EXCLUSIVE_EXCLUSIVE
        )

        // 3
        val spannable3 =
            SpannableStringBuilder(context.getString(R.string.app_agreement_protection_tip3))
        spannable3.setSpan(
            LeadingMarginSpan.Standard(0, 32),
            0,
            spannable3.length,
            Spannable.SPAN_EXCLUSIVE_EXCLUSIVE
        )

        // 4
        val spannable4 =
            SpannableStringBuilder(context.getString(R.string.app_agreement_protection_tip4))
        val protocolClickableSpan3: ClickableSpan = object : ClickableSpan() {
            override fun onClick(widget: View) {
                PagePilotManager.pageWebView(URLStatics.userAgreementURL)
            }
        }
        val protocolClickableSpan4: ClickableSpan = object : ClickableSpan() {
            override fun onClick(widget: View) {
                PagePilotManager.pageWebView(URLStatics.privacyAgreementURL)
            }
        }
        spannable4.setSpan(protocolClickableSpan3, 9, 15, Spannable.SPAN_EXCLUSIVE_EXCLUSIVE)
        spannable4.setSpan(spanColor, 9, 15, Spannable.SPAN_EXCLUSIVE_EXCLUSIVE)
        spannable4.setSpan(protocolClickableSpan4, 16, 22, Spannable.SPAN_EXCLUSIVE_INCLUSIVE)
        spannable4.setSpan(spanColor, 16, 22, Spannable.SPAN_EXCLUSIVE_INCLUSIVE)
        // 5
        val spannable5 =
            SpannableStringBuilder(context.getString(R.string.app_agreement_protection_tip5))
        val boldSpan = StyleSpan(Typeface.BOLD)
        spannable5.setSpan(boldSpan, 0, spannable5.length, Spannable.SPAN_INCLUSIVE_EXCLUSIVE)
        val finalText = TextUtils.concat(
            spannable1,
            "\n\n",
            spannable2,
            "\n",
            spannable3,
            "\n\n",
            spannable4,
            "\n\n",
            spannable5
        )
        binding.tvProtection.text = finalText
        binding.btnAgree.setOnClickListener { view -> getOnButtonClickListener().onRightButtonClick() }
        binding.tvProtection.movementMethod = LinkMovementMethod.getInstance()
    }

    override fun setGravity() {
        window?.let { window ->
            window.setBackgroundDrawableResource(android.R.color.transparent)
            window.setDimAmount(0f)
            window.setType(WindowManager.LayoutParams.TYPE_APPLICATION_PANEL)
            window.setFlags(
                WindowManager.LayoutParams.FLAG_LAYOUT_IN_SCREEN, WindowManager.LayoutParams.FLAG_LAYOUT_IN_SCREEN
            )
            window.setFlags(
                WindowManager.LayoutParams.FLAG_LAYOUT_NO_LIMITS, WindowManager.LayoutParams.FLAG_LAYOUT_NO_LIMITS
            )
            window.setLayout(
                ViewGroup.LayoutParams.MATCH_PARENT,
                ViewGroup.LayoutParams.MATCH_PARENT
            )
            window.attributes.gravity = Gravity.CENTER
        }
    }
}